### PR TITLE
fix(rules): handle anonymous class

### LINKED ISSUE #422: Error while parsing TS file with mixin
I get this error, 

    TypeError: Cannot read property 'text' of undefined at /home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/noAttributeParameterDecoratorRule.js:52:42
        at /home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/util/function.js:26:59
        at Maybe.bind (/home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/util/function.js:21:20)
        at Maybe.fmap (/home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/util/function.js:26:21)
        at /home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/noAttributeParameterDecoratorRule.js:50:14
        at Object.validate (/home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/walkerFactory/walkerFn.js:18:82)
        at /home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/walkerFactory/walkerFn.js:53:27
        at Array.forEach (native)
        at class_1.visitNode (/home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/codelyzer/walkerFactory/walkerFn.js:51:28)
        at /home/UX/arakovskyux/workspaces/pvzp-clfe/node_modules/tslint/lib/language/walker/syntaxWalker.js:535:63`

when my code contains declaration of a mixin
(https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#typescript-22)

E.g. something like this

    type Constructor<T> = new (...args: any[]) => T;

    export function MyUtils<T extends Constructor<{}>>(Base: T) {
      return class extends Base {
        public myMethod(a, b): boolean {
          ....
        }
      };
    }

It may be related to issue #28.

I am using latest version 3.2.0

### PR DESCRIPTION
**Pull request by mgechev**
Fix #422

### CODE REVIEW

**Comment by wKoza** [comment]
@mgechev , travis complains because of linting error :

> ERROR: test/noAttributeParameterDecoratorRule.spec.ts[17, 12]: Missing semicolon

**Comment by mgechev** [comment]
@wKoza that's for the catch. Should be fine now.

File: src/noAttributeParameterDecoratorRule.ts [hunk 1]
```diff
@@ -29,9 +29,9 @@ export class Rule extends Lint.Rules.AbstractRule {
             const syntaxKind = SyntaxKind.current();
             return Maybe.lift(node.parent)
                 .fmap(parent => {
-                    if (parent.kind === syntaxKind.ClassExpression) {
+                    if (parent.kind === syntaxKind.ClassExpression && parent.parent.name) {
```
Comment (wKoza) [inline src/noAttributeParameterDecoratorRule.ts]: parent.parent is always defined ?

Comment (mgechev) [reply src/noAttributeParameterDecoratorRule.ts]: Yes, the constructor is always in a class.

**Pull request merged**